testcase/testcase.go

```diff
@@ -254,6 +254,20 @@ func (tcs *TestCaseSet) Compare(events []logstash.Event, diffCommand []string, l
 		}
 	}()
 
+	// Sort the events and expected events so that they are both in the same order
+	// This is important as event ordering is not guaranteed under logstash config changes
+	sort.Slice(events, func(i,j int) bool {
+		iText, _ := json.Marshal(events[i])
+		jText, _ := json.Marshal(events[j])
+		return string(iText) < string(jText)
+	})
+
+        sort.Slice(events, func(i,j int) bool {
```

Comment:
Did you mean tcs.ExpectedEvents here? Lines 259-263 and 265-269 are identical.

Should perhaps make a copy of tcs.ExpectedEvents and sort the copy. Or, sort the expected events when loading the test case file?

Finally, this patch hunk hasn't been gofmt'd. Travis detected this in https://travis-ci.org/github/magnusbaeck/logstash-filter-verifier/builds/731212029 but for some reason this wasn't reported back to the PR. I recently pushed a commit that upgrades golangci-lint so it'd be a good idea to rebase your branch.

Reply:
Should all be fixed now.  Thanks for spotting these issues.